include/ginkgo/core/matrix/coo.hpp

```diff
@@ -81,12 +81,15 @@ class Coo : public EnableLinOp<Coo<ValueType, IndexType>>,
             public ConvertibleTo<Dense<ValueType>>,
             public DiagonalExtractable<ValueType>,
             public ReadableFromMatrixData<ValueType, IndexType>,
-            public WritableToMatrixData<ValueType, IndexType> {
+            public WritableToMatrixData<ValueType, IndexType>,
+            public EnableAbsoluteComputation<
+                remove_complex<Coo<ValueType, IndexType>>> {
     friend class EnableCreateMethod<Coo>;
     friend class EnablePolymorphicObject<Coo, LinOp>;
     friend class Csr<ValueType, IndexType>;
     friend class Dense<ValueType>;
     friend class CooBuilder<ValueType, IndexType>;
+    friend class Coo<to_complex<ValueType>, IndexType>;
```

Comment:
nit: Consistency for the same thing would be nice here:
either use `remove_complex`, or `to_complex` for the complex type in this file for the absolute type (If we ever have different implementations for `to_complex` and `remove_complex`, this would result in a compile time error).
Same for the `using absolute_type` below.

Reply:
I do not fully understand it. what's the consistency here?

Reply:
Do you mean to use only `remove/to_complex<Class>`  or only `Class<remove/to_complex<type>>` in the file?

Reply:
I don't think he can make it consistent since friendship works in reverse esp. with the EnableAbsolute. He needs to remove the complex from the type given to the EnableAbsolute mixin (because the return type will be the non complex variant, the one specified here). On the other hand, that non complex variant needs to provide friendship access to the complex version (and not the other way around).
The only way to make it the same is to not give the return type to the EnableAbsolute mixin but the input type. AFAIK though this version is less generic.

Reply:
My bad, I misread `to_complex` and for some reason thought it was `to_real`.
I will mark this as resolved.